pkg/vpcmodel/drawioOutput.go

```diff
@@ -134,30 +141,37 @@ func (d *DrawioOutputFormatter) showResource(res DrawioResourceIntf) bool {
 	return d.uc != AllSubnets || res.ShowOnSubnetMode()
 }
 
-func (d *DrawioOutputFormatter) WriteOutput(c1, c2 *VPCConfig,
-	conn *VPCConnectivity,
-	subnetsConn *VPCsubnetConnectivity,
+func (d *DrawioOutputFormatter) WriteOutput(c1, c2 map[string]*VPCConfig,
+	conn map[string]*VPCConnectivity,
+	subnetsConn map[string]*VPCsubnetConnectivity,
 	cfgsDiff *diffBetweenCfgs,
 	outFile string,
 	grouping bool,
-	uc OutputUseCase) (*SingleAnalysisOutput, error) {
+	uc OutputUseCase) (string, error) {
 	switch uc {
 	case AllEndpoints:
-		var gConn *GroupConnLines
-		if conn != nil {
-			gConn = conn.GroupedConnectivity
+		gConn := map[string]*GroupConnLines{}
+		for name, vpcConn := range conn {
+			gConn[name] = vpcConn.GroupedConnectivity
 		}
 		d.init(c1, gConn, uc)
 	case AllSubnets:
-		var gConn *GroupConnLines
+		gConfigs := map[string]*VPCConfig{}
+		gConn := map[string]*GroupConnLines{}
 		if subnetsConn != nil {
-			gConn = subnetsConn.GroupedConnectivity
+			for name, vpcConn := range subnetsConn {
+				gConn[name] = vpcConn.GroupedConnectivity
+				gConfigs[name] = vpcConn.VPCConfig
+			}
```

Comment:
Basically what you do here is transform the data to the format expected by drawio, correct?

Reply:
also choosing the right input:
in case of AllSubnets i take the vpcConn.VPCConfig to be the input, (and not c1)